exp/internal/controllers/machinepool_controller_noderef_test.go

```diff
@@ -175,13 +177,33 @@ func TestMachinePoolGetNodeReference(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:           "valid provider id, valid aws node, with minReadySeconds",
+			providerIDList: []string{"aws://us-east-1/id-node-1"},
+			expected: &getNodeReferencesResult{
+				references: []corev1.ObjectReference{{Name: "node-1"}},
+				available:  0,
+				ready:      1,
+			},
+			minReadySeconds: 20,
+		},
+		{
+			name:           "valid provider id, valid aws node, with minReadySeconds equals 0",
+			providerIDList: []string{"aws://us-east-1/id-node-1"},
+			expected: &getNodeReferencesResult{
+				references: []corev1.ObjectReference{{Name: "node-1"}},
```

Comment:
As mentioned in a previous commen. Let's start validating expected.available/ready against result (somewhere ~ in l.219)

Today this test never validates the available and ready fields.

In fact. At the moment they are always 0 because we never set conditions correclty to get ready or available Nodes. Let's change that as well.

(otherwise we don't have any test coverage for ready/available)

Reply:
I wasn't able to follow the force-pushes but I see that the test is now validating available and ready fields based on a different minReadySeconds. That looks good!!

I think the only thing missing is a test case when the NodeReady condition is false, both ready and available should be 0.